auto allocate directory in local and external cluster

UmmulkiramR opened the pull request:
Reference ticket: [#33](https://github.com/icgc-argo/workflow-management-scheduler/issues/33)

The scheduler will allocate work directories in the external cluster to the incoming workflow jobs and also inform management about the cluster in which to fire the jobs. Work directory allocation is still based on the cost calculation performed by the Scheduler

Commit:
auto allocate directory in local and external cluster

joneubank commented on src/main/java/org/icgc/argo/workflow_management_scheduler/components/DirScheduler.java:
```diff
@@ -110,28 +110,31 @@ private List<Run> getNextScheduledRuns(List<Run> activeRuns, List<Run> queuedRun
                             nextRunToInit.getWorkflowEngineParams().getWorkDir(),
                             config.getWorkDirTemplate(),
-                            value);
+                            value.split(":")[0]);
```
We have `":"` hardcoded in quite a few places here. Can we introduce a global constant for `DIRECTORY_STRING_DELIMITER = ":"` or some other meaningful name that can be referenced throughout instead of hardcoding this string repeatedly?

It may even be better to have a function that splits a string from `nfs-external/nfs-1:cluster2` into an object with a directory and cluster property. Ideally we would put the logic for splitting a string into these properties in one place and not repeat the split logic throughout the code.

joneubank reviewed (commented):
I feel like I am missing a critical piece of understanding here... I came in expecting to find logic that would find compare the capacity of a directory to the cost of running another job there. If I am interpreting this correctly, that logic exists and the only difference from this change is that it will now consider directories from multiple clusters when performing the existing calculation. The changes here are for helping the scheduler configure the correct cluster based on the directory it chooses.

If that is correct, then this is a nice solution in the sense that it has quite small impact on existing working logic.

Couple concerns I have:
1. Would like to see some new tests to ensure that the cluster is set correctly based on the selected workspace.
2. How do we handle the case where the config has invalid values? examples would be a directory missing the cluster name, or a cluster name that it doesn't have in the cluster config.

Commit:
Added
- check for missing cluster value for a directory
- the directory cluster delimiter

Commit:
refactored and formatted code

Commit:
updated tests to include the directory:cluster config
added new test to check for missing cluster
formatted code.

Commit:
updated cluster check
